railties/lib/rails/test_unit/minitest_plugin.rb

```diff
@@ -80,7 +80,8 @@ def self.plugin_rails_init(options)
       Minitest.backtrace_filter = ::Rails.backtrace_cleaner if ::Rails.respond_to?(:backtrace_cleaner)
     end
 
-    self.reporter.reporters.clear # Replace progress reporter for colors.
+    # Replace progress reporter for colors.
+    self.reporter.reporters.delete_if { |reporter| reporter.is_a?(SummaryReporter) || reporter.is_a?(ProgressReporter) }
```

Comment:
Yup, let's try only clearing Minitests reporters. But then we'd still need to use `kind_of?` so we won't remove custom subclasses.